std/StringTools.hx

```diff
@@ -39,105 +39,105 @@ class StringTools {
 	/**
 		Encode an URL by using the standard format.
 	**/
-	#if (!java && !cpp && !lua && !eval) inline #end public static function urlEncode( s : String ) : String {
+	#if (!java && !cpp && !lua && !eval) inline #end public static function urlEncode(s:String):String {
 		#if flash
-			return untyped __global__["encodeURIComponent"](s);
+		return untyped __global__["encodeURIComponent"](s);
 		#elseif neko
-			return untyped new String(_urlEncode(s.__s));
+		return untyped new String(_urlEncode(s.__s));
 		#elseif js
-			return untyped encodeURIComponent(s);
+		return untyped encodeURIComponent(s);
 		#elseif cpp
-			return untyped s.__URLEncode();
+		return untyped s.__URLEncode();
 		#elseif java
-			return postProcessUrlEncode(java.net.URLEncoder.encode(s, "UTF-8"));
+		return postProcessUrlEncode(java.net.URLEncoder.encode(s, "UTF-8"));
 		#elseif cs
-			return untyped cs.system.Uri.EscapeDataString(s);
+		return untyped cs.system.Uri.EscapeDataString(s);
 		#elseif python
-			return python.lib.urllib.Parse.quote(s, "");
+		return python.lib.urllib.Parse.quote(s, "");
 		#elseif hl
-			var len = 0;
-			var b = @:privateAccess s.bytes.urlEncode(len);
-			return @:privateAccess String.__alloc__(b,len);
+		var len = 0;
+		var b = @:privateAccess s.bytes.urlEncode(len);
+		return @:privateAccess String.__alloc__(b, len);
 		#elseif lua
-			s = lua.NativeStringTools.gsub(s, "\n", "\r\n");
-			s = lua.NativeStringTools.gsub(s, "([^%w %-%_%.%~])", function (c) {
-				return lua.NativeStringTools.format("%%%02X", lua.NativeStringTools.byte(c) + '');
-			});
-			s = lua.NativeStringTools.gsub(s, " ", "+");
-			return s;
+		s = lua.NativeStringTools.gsub(s, "\n", "\r\n");
+		s = lua.NativeStringTools.gsub(s, "([^%w %-%_%.%~])", function(c) {
+			return lua.NativeStringTools.format("%%%02X", lua.NativeStringTools.byte(c) + '');
+		});
+		s = lua.NativeStringTools.gsub(s, " ", "+");
+		return s;
 		#else
-			return null;
+		return null;
 		#end
 	}
 
-#if java
-	private static function postProcessUrlEncode( s : String ) : String {
+	#if java
+	private static function postProcessUrlEncode(s:String):String {
 		var ret = new StringBuf();
-		var i = 0,
-		    len = s.length;
+		var i = 0, len = s.length;
 		while (i < len) {
-			switch(_charAt(s, i++)) {
-			case '+'.code:
-				ret.add('%20');
-			case '%'.code if (i <= len - 2):
-				var c1 = _charAt(s, i++),
-				    c2 = _charAt(s, i++);
-				switch[c1, c2] {
-				case ['2'.code, '1'.code]:
-					ret.addChar('!'.code);
-				case ['2'.code, '7'.code]:
-					ret.addChar('\''.code);
-				case ['2'.code, '8'.code]:
-					ret.addChar('('.code);
-				case ['2'.code, '9'.code]:
-					ret.addChar(')'.code);
-				case ['7'.code, 'E'.code] | ['7'.code, 'e'.code]:
-					ret.addChar('~'.code);
-				case _:
-					ret.addChar('%'.code);
-					ret.addChar(cast c1);
-					ret.addChar(cast c2);
-				}
-			case var chr:
-				ret.addChar(cast chr);
+			switch (_charAt(s, i++)) {
+				case '+'.code:
```

Comment:
I'll open a discussion for this one. I think that we should keep the case on the same line as the switch.

Reply:
Resolving this since the discussion is in the main thread.